From 57cc686cc2a561d63b5779025e855053706176f0 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Tue, 12 Jan 2016 17:29:19 +0800
Subject: [PATCH 0839/5242] MLK-12215 video: mxsfb: free the 'timings' after
 using it to avoid resource leak

commit  38f989497034a28874f7ba08769979f644102d94 from
https://source.codeaurora.org/external/imx/linux-imx.git

The 'timings' is allocated by 'of_get_display_timings()'
dynamically. So it should be freed when it is not used
anymore.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxsfb.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/video/fbdev/mxsfb.c b/drivers/video/fbdev/mxsfb.c
index 727a3ec..601a27c 100644
--- a/drivers/video/fbdev/mxsfb.c
+++ b/drivers/video/fbdev/mxsfb.c
@@ -1105,7 +1105,7 @@ static int mxsfb_init_fbinfo_dt(struct fb_info *fb_info)
 	struct device_node *np = host->pdev->dev.of_node;
 	struct device_node *display_np;
 	struct device_node *timings_np;
-	struct display_timings *timings;
+	struct display_timings *timings = NULL;
 	const char *disp_dev;
 	u32 width;
 	int i;
@@ -1194,6 +1194,8 @@ static int mxsfb_init_fbinfo_dt(struct fb_info *fb_info)
 put_timings_node:
 	of_node_put(timings_np);
 put_display_node:
+	if (timings)
+		kfree(timings);
 	of_node_put(display_np);
 	return ret;
 }
-- 
1.7.9.5

